core/txpool/legacypool/legacypool.go

```diff
@@ -433,11 +434,11 @@ func (pool *LegacyPool) SetGasTip(tip *big.Int) {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
-	old := pool.gasTip.Load()
-	pool.gasTip.Store(new(big.Int).Set(tip))
+	old := pool.gasTip
+	pool.gasTip = uint256.MustFromBig(tip)
 
 	// If the min miner fee increased, remove transactions below the new threshold
-	if tip.Cmp(old) > 0 {
+	if old == nil || pool.gasTip.Cmp(old) > 0 {
```

Comment:
Can remove the defensive check `old == nil`.  we don't ever instantiate the txpool with a nil value for the tx price limit.